### PR TITLE
<MessageBox*/> - add footer actions and image support

### PR DESCRIPTION
**Pull request by liorMar**
### What changed

feat(MessageBoxFunctionalLayout) added image support, side action to the footer
feat(MessageBoxMarketerialLayout) added footer Bottom Children support, cross theme support

### Why it changed

Guild week 16

---

- [x] Change is tested
- [x] Change is documented
- [x] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### CODE REVIEW

File: src/MessageBox/FooterLayout.js [hunk 1]
```diff
@@ -13,14 +13,15 @@ const FooterLayout = ({
...
   buttonsHeight,
   enableOk,
-  enableCancel
-}) =>
+  enableCancel,
+  sideActions
+}) => (
   <div>
     <div className={styles.footer} data-hook="message-box-footer">
+      {sideActions && <div className={styles.footeractions} children={sideActions}/>}
```
Comment (shlomitc) [inline src/MessageBox/FooterLayout.js]: what is `footeractions`?  where does it come from?

Comment (liorMar) [reply src/MessageBox/FooterLayout.js]: it the extra element you add to the footer
screen shot 2018-08-19 at 19 29 25

**Comment by liorMar** [comment]
same PR

File: src/MessageBox/FooterLayout.js (hunk 1 above)
Comment (shlomitc) [reply src/MessageBox/FooterLayout.js]: `FooterLayout.scss` doesn't contain `footeractions` class name, what is the effect of it?

File: src/MessageBox/FooterLayout.scss [hunk 2]
```diff
@@ -20,6 +20,8 @@
 
 .footerbuttons {
   margin-left: auto;
+  flex-shrink: 0;
+  padding-left: 24px;
```
Comment (shlomitc) [inline src/MessageBox/FooterLayout.scss]: should it always have padding left or only when the side actions are added?

File: stories/MessageBox/Alert.js [hunk 3]
```diff
@@ -15,12 +15,18 @@ import SecondaryAlertRaw from '!raw-loader!./AlertExamples/Secondary';
 import FootNoteAlert from './AlertExamples/FootNote';
 import FootNoteAlertRaw from '!raw-loader!./AlertExamples/FootNote';
 
+import Image from './AlertExamples/Image';
```
Comment (shlomitc) [inline stories/MessageBox/Alert.js]: Add example also for the use case of both Image and side actions

File: stories/MessageBox/Announcement.js [hunk 4]
```diff
@@ -9,6 +9,9 @@ import CodeExample from 'wix-storybook-utils/CodeExample';
 import AnnouncementStandard from './AnnouncementExamples/Standard';
 import AnnouncementStandardRaw from '!raw-loader!./AnnouncementExamples/Standard';
 
+import AnnouncementCrossTheme from './AnnouncementExamples/CrossTheme';
```
Comment (shlomitc) [inline stories/MessageBox/Announcement.js]: Please call it `Premium Action` as described in the Zeplin

File: src/MessageBox/FooterLayout.js [hunk 5]
```diff
@@ -61,7 +64,8 @@ FooterLayout.propTypes = {
   theme: PropTypes.string,
   buttonsHeight: PropTypes.string,
   children: PropTypes.any,
-  bottomChildren: PropTypes.node
+  bottomChildren: PropTypes.node,
+  sideActions: PropTypes.node
```
Comment (shlomitc) [inline src/MessageBox/FooterLayout.js]: please add this to the `README.md` file

File: src/MessageBox/MessageBoxMarketerialLayout.driver.js [hunk 6]
```diff
@@ -18,6 +18,7 @@ const messageBoxMarketerialLayoutDriverFactory = ({element, wrapper, component})
     closeMessageBox: () => ReactTestUtils.Simulate.click(closeButton()),
     getTitle: () => element.querySelector('[data-hook="message-box-title"]').textContent,
     getContentBySelector: selector => element.querySelector(selector),
+    getChildBySelector: selector => element.querySelector(selector),
```
Comment (shlomitc) [inline src/MessageBox/MessageBoxMarketerialLayout.driver.js]: are you sure this is necessary for the driver?

File: src/MessageBox/MessageBoxFunctionalLayout.scss [hunk 7]
```diff
@@ -35,6 +35,21 @@ $minimum-modal-margin: 48px;
   }
 }
 
+.message-with-image {
+  display: flex;
+
+  .image {
+    display: inline-block;
+    margin: -12px 30px -42px 0;
```
Comment (shlomitc) [inline src/MessageBox/MessageBoxFunctionalLayout.scss]: did you consider putting the `image` alongside the body content and not inside of it?
This might drop the need for negative margins which is hard to follow, and just add margin/padding on demand when an image is needed

File: src/MessageBox/MessageBoxMarketerialLayout.scss [hunk 8]
```diff
@@ -118,3 +118,15 @@ $headerImageHeight: 185px;
 .fixedFooter {
   flex-shrink: 0;
 }
+
+.bottomChildren {
```
Comment (shlomitc) [inline src/MessageBox/MessageBoxMarketerialLayout.scss]: It looks like code duplication to `FooterLayout.scss` - is that by intention?
I suggest refactoring it to a common place

Comment (liorMar) [reply src/MessageBox/MessageBoxMarketerialLayout.scss]: is it OK to have MessageBoxCommon.scss file ?

Comment (shlomitc) [reply src/MessageBox/MessageBoxMarketerialLayout.scss]: Yes

File: src/MessageBox/FooterLayout.scss (hunk 2 above)
Comment (liorMar) [reply src/MessageBox/FooterLayout.scss]: done

File: stories/MessageBox/Alert.js (hunk 3 above)
Comment (liorMar) [reply stories/MessageBox/Alert.js]: done

File: stories/MessageBox/Announcement.js (hunk 4 above)
Comment (liorMar) [reply stories/MessageBox/Announcement.js]: done

File: src/MessageBox/FooterLayout.js (hunk 5 above)
Comment (liorMar) [reply src/MessageBox/FooterLayout.js]: done

File: src/MessageBox/MessageBoxMarketerialLayout.driver.js (hunk 6 above)
Comment (liorMar) [reply src/MessageBox/MessageBoxMarketerialLayout.driver.js]: done

File: src/MessageBox/MessageBoxMarketerialLayout.scss (hunk 8 above)
Comment (liorMar) [reply src/MessageBox/MessageBoxMarketerialLayout.scss]: done

**Comment by shlomitc** [comment]
@LittleTal - Is this color available in the `Text` component?
https://wix-wix-style-react.surge.sh/?selectedKind=1.%20Foundation&selectedStory=1.2%20Text&full=0&addons=0&stories=1&panelRight=0

If so - which of the combinations should we use? (light? secondary?)

**Comment by LittleTal** [comment]
@shlomitc  
MessageBoxFunctionalLayout checkbox - T1 (16px)
MessageBoxMarketerialLayout footer text - T3.1 (14px)

**Comment by liorMar** [comment]
@LittleTal @shlomitc changed texts color/size

**Comment by liorMar** [comment]
@shlomitc lint fixed
